### ISSUE TITLE
jakt transpiles with incorrect c_char value resulting in clang error

### ISSUE THREAD

**Issue by demizer** [SerenityOS/jakt#276]
I was using c_char wrong but jakt eagerly attempted to compile:

```
$ cat main.jakt 
function main() {
    let str_c: c_char = '\n'
    print("{}", str_c)
}
```
clang error output:

```
$ ./build.sh && ./a.out
output.cpp:9:28: warning: missing terminating ' character [-Winvalid-pp-token]
        const char str_c = '\';
                           ^
output.cpp:9:28: error: expected expression
1 warning and 1 error generated.
```

**Comment by mustafaquraish** [comment]
The issue here seems more to me like the escape characters are not handled. This example works fine with:
```
function main() {
    let str_c: c_char = 'n'
    print("{}", str_c)
}
```

**Comment by mustafaquraish** [comment]
Fixed in 334191e
